snap/hooks/configure

```diff
@@ -7,3 +7,6 @@ k8s::common::setup_env
 k8s::cmd::k8s x-snapd-config reconcile
 
 k8s::containerd::ensure_systemd_defaults
+
+# Sanitize kube-apiserver feature gates that don't exist in k8s 1.33
+k8s::apiserver::sanitize_feature_gates "DetectCacheInconsistency" "ListFromCacheSnapshot" "SizeBasedListCostEstimate"
```

Comment:
i think i understand that this removes these feature gates (on the 1.33 snap) if they've been set by a previous install (previously 1.33) 

This is similar to giving a caveman instructions on which tools to avoid when repairing his tractor.  Doesn't it feel odd to tell a 1.33 cluster to avoid a specific set of 1.34 features?

I know i'm not offering an improvement here.  I guess there's no way to query from the 1.33 executable a list of "Available" features to enabled, and remove any that are no longer in that list of available features?


```
/snap/k8s/current/bin/kube-apiserver --help | awk '/^ *kube:/{print $1}'
```

provides a pretty compelling list -- but i'm not sure it's complete.

Reply:
Good suggestion!